typescript/utils/src/objects.ts

```diff
@@ -27,8 +23,8 @@ export function objMap<
   K extends keyof M,
   O,
   I = ValueOf<M>,
->(obj: M, func: (k: K, v: I) => O): MappedObject<M, O> {
-  return Object.fromEntries<O>(objMapEntries(obj, func)) as MappedObject<M, O>;
+>(obj: M, func: (k: K, v: I) => O): Record<K, O> {
+  return Object.fromEntries<O>(objMapEntries(obj, func)) as Record<K, O>;
```

Comment:
I'm not sure why we were using `MappedObject` before, this simpler version returns more accurate types from objMap calls